Gruppe.A1.End-to-End_Encryption_mit_IPv6_und_integriertem_IPSec.ThushjandanPonnudurai.PascalBaumann.Termpaper.tex

```diff
@@ -508,12 +508,39 @@ \section{Experiment results}
+
+Firstly we have to automate the certificate enrollment process. Unfortunately it is not easily possible to realize this task with OpenSSL as CA. 
+Therefore Microsoft Active Directory Certificate Services (ADCS) will be a better choice than OpenSSL. 
+One of the main reasons is that ADCS can be perfectly integrated in an Active Directory environment. Certificate Autoenrollment can be easily configured with group policies for Windows computers. 
+Another reason for ADCS is that it supports the Simple Certificate Enrollment Protocol (SCEP).
+With SCEP it is possible to enroll automatically certificates for network devices and also for Linux machines. 
+Network Device Enrollment Service (NDES) is a module of ADCS. It implements the SCEP protocol. 
+
+Now how we should deploy the IPsec configuration to clients? With group policies we could easily enroll certificates and configure IPsec connection profiles for Windows machines. 
+After that every Windows computer will be able to establish IPsec tunnels to all hosts that are specified in the IPsec connection profile.
+
+Deploying IPsec profiles to Linux machines is a little bit trickier. We could use a software configuration management tool like Ansible. 
+With Ansible and additional custom scripts it is possible to automate the certificate enrollment and renewal process (Using a SCEP client).
+Also the installation and configuration of the IPsec client could be automated with Ansible. 
+
+There are existing a couple of SCEP clients for Linux. StrongSwan is delivering a client named "ipsec\_scepclient" \parencite{Hutter2005} or CertNanny \parencite{Bartosch2017}, 
+which uses the SSCEP Tool from Jarkko Turkulainen, is also a good candidate for certificate autoenrollment.
+
+Combining these given tools it is possible to build up automated End-to-End encryption as well for IPv6-only Linux environments.
 
 \section{Conclusion}
 \label{sec:Conc}
-% Tush : WRITE CONCLUSION
+In section \ref{sec:ExpectedResult} we adjusted to have difficulties configuring IPsec on a Windows computer. But the contrary has been happened during the experiment. 
+We had more difficulties to configure IPsec VPN on Linux than on Windows. Windows doesn't support a lot of encryption and hashing algorithm, but nevertheless it was easily to find the matching algorithm for both clients.
+
+End-to-End encryption can be in a Windows IPv6-only network easily deployed. Microsoft is delivering already suitable tools for this use case. 
+If Linux machines are also a part of the network, then several tools needs to be combined to achieve this goal. Thus it is possible to realize an End-to-End encryption between managed-machines in a network. 
+We have investigated only solutions with managed machines. It could be possible to offer End-to-End encryption for unmanaged clients, but this was not the scope of our term paper
```

Comment:
We have investigated a solution with managed machines only. It _would_ be possible to offer End-to-End encryption for clients with a centralized configuration, but this was not _in_ the scope of our term paper.